src/strcat.rs

```diff
@@ -0,0 +1,29 @@
+//! Rust implementation of C library function `strcat`
+//!
+//! Licensed under the Blue Oak Model Licence 1.0.0
+
+use crate::{strcpy, strlen, CChar};
+
+/// Rust implementation of C library function `strcat`. Passing NULL
+/// (core::ptr::null()) gives undefined behaviour.
+#[no_mangle]
+pub unsafe extern "C" fn strcat(dest: *mut CChar, src: *const CChar) -> *const CChar {
+	strcpy(dest.add(strlen(dest)), src);
```

Comment:
You can use the path-qualified version of the function inside the function body. Then you don't have to gate it on the features being enabled.